docs/Contributors_Guide/github_workflow.rst

```diff
@@ -3,246 +3,349 @@
+    upstream	https://github.com/dtcenter/METplus (fetch)
+    upstream	https://github.com/dtcenter/METplus (push)
+
+  where <your-github-user> is your GitHub username.
+
+
+Generate a feature branch from the 'develop' branchfor new development
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+* Create a feature branch in the NCAR/METplus GitHub repository following this naming convention
+
+| *feature_<Github Issue number>_<brief_description>*
 |
+
+  For example, for GitHub issue #777 that creates new wrapper xyz, the feature branch would be named:
+
+| *feature_777_wrapper_xyz*
 |
-| *Re-cap*:
-|   You've created a feature branch, made changes, committed those changes to the repository, pushed them to GitHub,
-|   opened a pull request,had your changes merged by the repository maintainers, and finally performed some clean-up.
+
+* Create your feature branch based off the develop branch
+
+  .. code-block:: none
+
+    git checkout develop
+
+* Verify that you are currently working in the develop branch by running
+
+  .. code-block:: none
+
+    git branch
+
+  You should see something like the following:
+
+  .. code-block:: none
+
+    * develop
+    main_v3.1
+
+  The asterisk, "*", indicates the currently active branch.
+
+* At the command line, create and checkout the feature branch. For example:
+
+  .. code-block:: none
+		  
+    git checkout -b feature_777_wrapper_xyz
+
+  replacing *feature_777_wrapper_xyz* with your feature branch name.
+
+* Verify that you are working in the correct branch by running:
+
+  .. code-block:: none
+
+    git branch
+
+  You should see something like the following:
+
+  .. code-block:: none
+		  
+    develop
+    main_v3.1
+    * feature_777_wrapper_xyz
+
+  where the asterisk, "*", indicates which branch is currently in use/checked out.
+
+
+Make changes to code in the feature branch
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Users should make changes to their feature branch and commit changes to their
+local repository.
+
+* Create code following the coding standards in the Coding Standards section of
+  the Contributor's Guide.
+
+* Provide some tests for your code using the pytest framework, provide user documentation
+  describing what the code does, and provide any necessary data.
+
+* Keep your fork in sync. While working, it is highly likely that changes are occurring in
+  the original repository.  This may impact your work.  Regularly use the following commands
+  to keep your fork in sync with the original repository.
+
+  .. code-block:: none
+		  
+    git pull upstream develop
+    git push origin develop
+
+  The first command pulls changes from the original repository (the
+  `METplus GitHub repository <https://github.com/dtcenter/METplus>`_ that you see when you
+  run *git remote -v* and that you set to upstream in step 4 above).  The second command
+  pushes those changes to your forked repository.
+
```

Comment:
Consider adding instructions to merge the local develop branch into the feature branch, i.e.

`git merge origin develop`

Reply:
Done!